core/src/main/java/hudson/PluginManager.java

```diff
@@ -1587,12 +1587,13 @@ public HttpResponse doUpdateSources(StaplerRequest req) throws IOException {
 
         if (req.hasParameter("remove")) {
             UpdateCenter uc = Jenkins.get().getUpdateCenter();
-            BulkChange bc = new BulkChange(uc);
-            try {
-                for (String id : req.getParameterValues("sources"))
-                    uc.getSites().remove(uc.getById(id));
-            } finally {
-                bc.commit();
+            try (BulkChange bc = new BulkChange(uc)) {
```

Comment:
Would not all calls to `abort()` be no-ops due to the use of `commit()` in a finally block?